steward/src/cellars/cellar_v2.rs

```diff
@@ -0,0 +1,702 @@
+
+                            debug!("encoded: {:?}", hex::encode(&swap_params));
+                            let call = steward_abi::compound_c_token_adaptor::SwapCall {
+                                asset_in: sp_call_parse_address(p.asset_in)?,
+                                asset_out: sp_call_parse_address(p.asset_out)?,
+                                amount_in: string_to_u256(p.amount_in)?,
+                                exchange: convert_exchange(p.exchange),
+                                params: swap_params.into(),
+                            };
+                            calls.push(CompoundCTokenAdaptorCalls::Swap(call).encode().into())
+                        }
+                        compound_c_token_adaptor::Function::OracleSwap(p) => {
+                            let oracle_swap_params =
+                                encode_oracle_swap_params(p.params.ok_or_else(|| {
+                                    ErrorKind::SPCallError.context("swap params cannot be empty")
+                                })?)?;
+
+                            debug!("encoded: {:?}", hex::encode(&oracle_swap_params));
+                            let call = steward_abi::compound_c_token_adaptor::OracleSwapCall {
+                                asset_in: sp_call_parse_address(p.asset_in)?,
+                                asset_out: sp_call_parse_address(p.asset_out)?,
+                                amount_in: string_to_u256(p.amount_in)?,
+                                exchange: convert_exchange(p.exchange),
+                                params: oracle_swap_params.into(),
+                                slippage: p.slippage,
+                            };
+                            calls.push(CompoundCTokenAdaptorCalls::OracleSwap(call).encode().into())
+                        }
+                        compound_c_token_adaptor::Function::RevokeApproval(p) => {
+                            let call = steward_abi::compound_c_token_adaptor::RevokeApprovalCall {
+                                asset: sp_call_parse_address(p.asset)?,
+                                spender: sp_call_parse_address(p.spender)?,
+                            };
+                            calls.push(
+                                CompoundCTokenAdaptorCalls::RevokeApproval(call)
+                                    .encode()
+                                    .into(),
+                            )
+                        }
+                    }
+                }
+            }
+            CallData::VestingSimpleCalls(params) => {
+                for c in params.calls {
+                    match c.function.expect("adaptor function was empty") {
+                        vesting_simple_adaptor::Function::DepositToVesting(p) => {
+                            let call = steward_abi::vesting_simple_adaptor::DepositToVestingCall {
+                                vesting_contract: sp_call_parse_address(p.vesting_contract)?,
```

Comment:
Similar to `CErc20`, the actual input in the contract here is a `VestingSimple` rather than an `ERC20`...again, the ABI generated code indicates this should work, just wondering if we also want to include a test in the mock contract where these types are imported and used.

Reply:
My thinking was since the ABI defines them as the same underlying type, I didn't want to set a precedent of digging up solidity code to import into the mock contract every time an adaptor does this.